### PR TITLE
Ledger-API Conformance test for V0 Contract IDs

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Runner.scala [hunk 1]
```diff
@@ -85,7 +85,12 @@ final class Runner[T <: ReadWriteService, Extra](
     )
     implicit val materializer: Materializer = Materializer(actorSystem)
 
-    val sharedEngine = new Engine(EngineConfig(config.allowedLanguageVersions))
+    val sharedEngine = new Engine(
+      EngineConfig(
+        config.allowedLanguageVersions,
+        forbidV0ContractId = true,
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Runner.scala:91]: we forbid V0 Contract ID in KV.

File: ledger/sandbox/src/main/scala/platform/sandboxnext/Runner.scala [hunk 2]
```diff
@@ -83,6 +83,7 @@ class Runner(config: SandboxConfig) extends ResourceOwner[Port] {
       allowedLanguageVersions = languageVersions,
       profileDir = config.profileDir,
       stackTraceMode = config.stackTraces,
+      forbidV0ContractId = true,
```
Comment (remyhaemmerle-da) [inline ledger/sandbox/src/main/scala/platform/sandboxnext/Runner.scala:86]: we also forbid V0 Contract ID in sandbox next.

File: ledger/ledger-api-test-tool-on-canton/BUILD.bazel [hunk 3]
```diff
@@ -149,7 +149,7 @@ conformance_test(
     server = ":canton-test-runner-with-dependencies",
     test_tool_args = [
         "--verbose",
-        "--include=ContractIdIT:RejectNonSuffixedCid,ContractIdIT:AcceptSuffixedCid",
+        "--include=ContractIdIT:RejectV0,ContractIdIT:RejectNonSuffixedV1Cid,ContractIdIT:AcceptSuffixedV1Cid",
```
Comment (remyhaemmerle-da) [inline ledger/ledger-api-test-tool-on-canton/BUILD.bazel:152]: @oliverse-da Canton test need to need to be  update

**Review by cocreature — APPROVED** [review]
Thanks!

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Runner.scala (hunk 1 above)
Comment (cocreature) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Runner.scala:91]: I think we might need to set it somewhere for pkvutils separately. I’m not sure it goes via this runner. Maybe we should turn on the rejectv0 tests by default to makes sure we notice if we forget to do this in pkvutils.

Comment (gerolf-da) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Runner.scala:91]: @miklos-da, FYI.

**Review by gerolf-da — APPROVED** [review]
👍

**Pull request merged**